x-pack/test/security_solution_endpoint/apps/endpoint/resolver.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common', 'timePicker', 'hosts', 'settings']);
+  const testSubjects = getService('testSubjects');
+  const esArchiver = getService('esArchiver');
+
+  describe('Endpoint Alert Resolver', function () {
+    this.tags(['ciGroup7']);
+
+    before(async () => {
+      await esArchiver.load('endpoint/resolver_tree/api_feature', { useCreate: true });
+      await pageObjects.hosts.navigateToSecurityHostsPage();
+      await pageObjects.common.sleep(4000);
+      const fromTime = 'Jan 1, 2018 @ 00:00:00.000';
+      const toTime = 'now';
+      await pageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+      await (await testSubjects.find('draggable-content-host.name')).click();
+      await testSubjects.existOrFail('header-page-title');
+      await (await testSubjects.find('navigation-events')).click();
+      await testSubjects.existOrFail('events-viewer-panel');
+      await pageObjects.common.sleep(4000);
+      await (await testSubjects.findAll('investigate-in-resolver-button'))[0].click();
+      await pageObjects.common.sleep(4000);
+    });
+
+    it('check that Resolver and Data table is loaded', async () => {
+      await testSubjects.existOrFail('resolver:graph');
+      await testSubjects.existOrFail('tableHeaderCell_name_0');
+      await testSubjects.existOrFail('tableHeaderCell_timestamp_1');
+    });
+
+    it('resolver Table and Node data same length', async () => {
+      const $1: string[] = [];
+      const $: string[] = [];
+
+      const tableData1 = await testSubjects.findAll('resolver:node-list:item');
+      for (const value of tableData1) {
+        const text = await value._webElement.getText();
+        $1.push(text.split('\n')[0]);
+      }
+      await testSubjects.click('resolver:graph-controls:zoom-out');
+      const Nodes = await testSubjects.findAll('euiButton__text');
```

Comment:
Instead of using the `data-test-subj` attributes from other projects, please use ones defined by Resolver. Our tests should not fail if another library changes its internal testing attributes. Also it is not clear (as a reader) what element you are interacting with. The button in the process node that contains the name of the process has a `data-test-subj` of `resolver:node:primary-button`. Consider using that instead.

Reply:
Done